docs/source/recipes/adding_rules.rst

```diff
@@ -102,14 +104,15 @@ Now, in a file named ``my_rules.py``, add
                 # To access config options, use self.rule
                 if document['username'] in self.rule['usernames']:
 
-                    # Convert the timestamp to a datetime object
-                    login_time = ts_to_dt(document['@timestamp'])
-
-                    # Convert the datetime object to a time string
-                    str_time = login_time.time().isoformat()
+                    # Convert the timestamp to a time object
+                    login_time = document['@timestamp'].time()
 
+                    # Convert time_start and time_end to time objects
+                    time_start = dateutil.parser.parse(self.rule['time_start']).time()
+                    time_end = dateutil.parser.parse(self.rule['time_end']).time()
+                    
                     # If the time falls between start and end
-                    if str_time > self.rule['time_start'] and str_time < self.rule['time_end']:
+                    if str_time > time_start and str_time < time_end:
```

Comment:
This is fine, but be cautious if you ever deal with representations of unixtime serialized through these time fields. If you do, you'll be a second off, due to not setting these as >= and <=.